examples/wandb_sweeps/train_wandb_sweep_agent.py

```diff
@@ -0,0 +1,178 @@
+    )
+    parser.add_argument(
+        "--save_top_k",
+        type=int,
+        default=defaults.SAVE_TOP_K,
+        help="Number of checkpoints to save. Default: %(default)s.",
+    )
+    parser.add_argument("--seed", type=int, help="Random seed")
+    # Needs to be true to log runtime metrics.
+    parser.add_argument(
+        "--wandb_logger",
+        action="store_true",
+        default=True,
+        help="Use Weights & Biases logging (log-in required). Default: True.",
+    )
+    return parser.parse_args()
+
+
+def run_train(args):
+    # First get trainer to initialize the wandb run
+    trainer = train._get_trainer_from_argparse_args(args)
+    pl.seed_everything(args.seed)
+    train_set, dev_set = train._get_datasets_from_argparse_args(args)
+    index = train.get_index(args.model_dir, args.experiment)
+    train_set.index.write(index)
+    util.log_info(f"Index: {index}")
+
+    # Model args come from the W&B sweep config.
+    kwargs = dict(wandb.config)
+    # Anything not specified in the config is taken from the CLI args.
+    kwargs.update({k: v for k, v in vars(args).items() if k not in kwargs})
+    train_loader, dev_loader = train.get_loaders(
+        train_set,
+        dev_set,
+        args.arch,
+        kwargs["batch_size"],
+        args.max_source_length,
+        args.max_target_length,
+    )
+    model = train.get_model(train_set, **kwargs)
+
+    # Train and log the best checkpoint.
+    best_checkpoint = train.train(
+        trainer, model, train_loader, dev_loader, args.train_from
+    )
+    util.log_info(f"Best checkpoint: {best_checkpoint}")
+
+    if args.predict:
+        # Deletes loaders to free those processes.
+        del train_loader
+        del dev_loader
+
+        # Fix a reasonable prediction batch_size.
+        pred_batch_size = 64
+        # Reloads the dev set.
+        loader = predict.get_loader(
+            dev_set,
+            args.arch,
+            pred_batch_size,
+            defaults.MAX_SOURCE_LENGTH,
+            defaults.MAX_TARGET_LENGTH
+        )
+        # Reloads the model.
+        model = predict.get_model(
+            args.arch,
+            train_set.config.has_features,
+            best_checkpoint,
+        )
+        # Hack to get the results dir for this version, from the checkpoint path
+        output = best_checkpoint.split("checkpoints")[0]
+        output = os.path.join(output, "predictions.tsv")
+        util.log_info(f"Writing predictions to {output}")
+        # Writes dev predictions with the best model.
+        predict.predict(trainer, model, loader, output)
+
+
+def main():
+    args = get_args()
+    try:
+        wandb.agent(
+            args.sweep_id,
+            function=functools.partial(run_train, args),
+            project=args.experiment,
+            count=args.max_num_runs
+        )
+    except Exception as e:
+        # Exits gracefully, so wandb logs the error
+        util.log_info(traceback.format_exc(), file=sys.stderr)
```

Comment:
`util.log_info` doesn't take a keyword arg and it already writes to stderr, just remove that.

Reply:
Oops! Yeah I was just find+replacing print...